lib/new-admin/graphql/modules/authentication.js

```
@@ -0,0 +1,223 @@
+const otplib = require('otplib')
+const bcrypt = require('bcrypt')
+
+const loginHelper = require('../../login')
+const T = require('../../../time')
+const users = require('../../../users')
+const sessionManager = require('../../../session-manager')
+
+const REMEMBER_ME_AGE = 90 * T.day
+
+function authenticateUser (username, password) {
+  return loginHelper.checkUser(username).then(hashedPassword => {
+    if (!hashedPassword) return null
```

You should throw error instead of returning null. We don't want to risk continuing the promise chain by mistake.